lib/internal/main/worker_thread.js

```diff
@@ -142,6 +142,9 @@ port.on('message', (message) => {
     const originalCwd = process.cwd;
 
     process.cwd = function() {
+      // SharedArrayBuffers can be disabled with --no-harmony-sharedarraybuffer.
+      if (typeof SharedArrayBuffer === 'undefined') return originalCwd();
```

Comment:
This is fine but we could prevent replacing `process.cwd` in the first place, so that the extra check is not needed when calling process.cwd().